### PR TITLE
add test according to #4246

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy [hunk 1]
```diff
@@ -428,4 +430,52 @@ class FlowStatSpec extends HealthCheckSpecification {
...
+        def traffGenSwitches = topology.activeTraffGens*.switchConnected*.dpId
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.src.dpId in traffGenSwitches && it.dst.dpId in traffGenSwitches
+        } ?: assumeTrue(false, "No suiting switches found")
+
+        and: "A flow with updated port on src endpoint via partial update"
+        def traffgenPortOnSrcSw = topology.activeTraffGens.find { it.switchConnected ==  switchPair.src}.switchPort
+        def srcFlowPort = (topology.getAllowedPortsForSwitch(topology.activeSwitches.find {
+            it.dpId == switchPair.src.dpId
+        }) - traffgenPortOnSrcSw).last()
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy:446]:
```suggestion
        def srcFlowPort = (topology.getAllowedPortsForSwitch(
            topology.find(switchPair.src.dpId)) - traffgenPortOnSrcSw).last()
```

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy [hunk 2]
```diff
@@ -428,4 +430,52 @@ class FlowStatSpec extends HealthCheckSpecification {
         cleanup:
         flow && flowHelperV2.deleteFlow(flow.flowId)
     }
+
+    @Tidy
+    def "System is able to collect stats after partial updating(port) on a flow endpoint"() {
+        given: "Two active neighboring switches connected to the traffgens"
+        def traffGenSwitches = topology.activeTraffGens*.switchConnected*.dpId
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy:438]: why only neighboring?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy [hunk 3]
```diff
@@ -428,4 +430,52 @@ class FlowStatSpec extends HealthCheckSpecification {
...
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildExam(northbound.getFlow(flow.flowId),
+                (int) flow.maximumBandwidth, 5)
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "System collects stats for ingress/egress cookies"
+        def flowInfo = database.getFlow(flow.flowId)
+        def tags = [switchid: switchPair.src.dpId.toOtsdFormat(), flowid: flow.flowId]
+        def metric = metricPrefix + "flow.raw.bytes"
+        def waitInterval = 10
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy:467]: can be way smaller. Not an impossible request for otsdb, but 10s wait is very significant for test duration

Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy]: `10s` was chosen just for reducing the amount of requests to opentsdb.
changed to `5s`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy]: thx, refactored

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/FlowStatSpec.groovy]: No special reason, 
it was the first thing that occurred to me.
refactored.

**Review by rtretyak — APPROVED** [review]

**Pull request merged**